### PR TITLE
✨ Added generic-autobumper to update all prow images

### PR DESCRIPTION
**Pull request by peppi-lotta**
Added automatic bumper to update all prow images to make developers lives easier. I've implemented the prow [Prow generic-autobumper](https://docs.prow.k8s.io/docs/components/cli-tools/generic-autobumper/) to achieve this. A prow job was defined to periodically(once a day) look for new images. When new images are found the auto bumper makes new a new pull request which can then just be approved. 

Once this is proven to work the current implementation for image updating can be removed. (This means the image list in prow/manifests/overlays/metal3/kustomizatin.yaml)

### CODE REVIEW

File: prow/manifests/overlays/metal3/external-plugins/generic-autobumper-config.yaml [hunk 1]
```diff
@@ -0,0 +1,136 @@
+gitHubToken: "/etc/github/token"
+skipPullRequest: false
+gitHubOrg: "metal3-io"
+gitHubRepo: "project-infra"
+remoteName: "project-infra"
+upstreamURLBase: "https://github.com/metal3-io/project-infra/main"
+includedConfigPaths:
+  - "prow"
```
Comment (tuminoid) [inline prow/manifests/overlays/metal3/external-plugins/generic-autobumper-config.yaml:8]: Use k8s yaml formatting, ie. no indentation on the lists.

File: prow/manifests/overlays/metal3/external-plugins/generic-autobumper-config.yaml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    refConfigFile: "prow/manifests/overlays/metal3/external-plugins/labels_cronjob.yaml" 
+    repo: "https://github.com/metal3-io/project-infra"
+    summarise: true
+    consistentImages: true
+  - name: "NeedsRebase"
+    prefix: "gcr.io/k8s-prow/needs-rebase:"
+    refConfigFile: "prow/manifests/overlays/metal3/external-plugins/needs-rebase_deployment.yaml" 
+    repo: "https://github.com/metal3-io/project-infra"
+    summarise: true
+    consistentImages: true
```
Comment (tuminoid) [inline prow/manifests/overlays/metal3/external-plugins/generic-autobumper-config.yaml:136]: Missing line feed.

File: prow/manifests/overlays/metal3/external-plugins/generic-autobumper-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+          readOnly: true
+        ports:
+        - containerPort: 8888
+      volumes:
+      - name: github-token
+        secret:
+          secretName: github-token
+      - name: config
+        configMap:
+          name: generic-autobumper-config
```
Comment (tuminoid) [inline prow/manifests/overlays/metal3/external-plugins/generic-autobumper-deployment.yaml:42]: Missing line feed.

**Review by tuminoid — COMMENTED** [review]
Nice improvement!

Minor formatting nits. Please also add some more documentation, and links  to the PR description as well and some expected behavior etc.

**Review by tuminoid — COMMENTED** [review]
/approve
/cc @lentzi90 
/hold
I assume we will run this in the cluster first, and then merge when its working.

File: prow/manifests/overlays/metal3/kustomization.yaml [hunk 4]
```diff
@@ -37,6 +38,10 @@ configMapGenerator:
   - labels.yaml
   name: label-config
   namespace: prow
+- files:
+  - config.yaml=external-plugins/generic-autobumper-deployment.yaml
```
Comment (lentzi90) [inline prow/manifests/overlays/metal3/kustomization.yaml:42]:
```suggestion
  - config.yaml=external-plugins/generic-autobumper-config.yaml
```

File: prow/manifests/overlays/metal3/external-plugins/generic-autobumper-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        - --dry-run=false
+        volumeMounts:
+        - name: github-token
+          mountPath: /etc/github
+          readOnly: true
+        - name: config
+          mountPath: /etc/config
+          readOnly: true
+        ports:
+        - containerPort: 8888
```
Comment (lentzi90) [inline prow/manifests/overlays/metal3/external-plugins/generic-autobumper-deployment.yaml:35]: We can remove this as it should not be listening on any ports
```suggestion
```

**Review by lentzi90 — COMMENTED** [review]
/approve
Great job!

**Comment by tuminoid** [comment]
/test-ubuntu-integration-main  
Random test, ignore

**Review by Rozzii — APPROVED** [review]
/hold cancel
/lgtm

**Pull request merged**